### ISSUE TITLE
Misleading comments in regexpfilterset

### ISSUE THREAD

**Issue by chenlujjj** [open-telemetry/opentelemetry-collector-contrib#30178]
### Component(s)

processor/filter

## Description

https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/internal/filter/filterset/regexp/regexpfilterset.go#L77

The comment of the regexp `Filterset.addFilters` is not consistent with its actual behavior.
```
// All regexes are automatically anchored to enforce full string matches.
```
From the code, the regular expressions are not anchored. If the user overlooks the code, he or she may misconfigure the filter rules. 

I noticed the "anchor" logic was removed in this [commit](https://github.com/open-telemetry/opentelemetry-collector-contrib/commit/71202e4553e23bc38cacae2fd674b1fa3e635248#diff-329734c26879e3f68e7b3d96a3dc2af0b182d6a1208a26cc16dc21e98fef1e0cL80-L82), just wondering why is was removed. 

### Collector version

0.79.0

## Environment
OS: (e.g., "Ubuntu 20.04")
Compiler(if manually compiled): (e.g., "go 14.2")

**Comment by crobert-1** [comment]
Hello @chenlujjj, it looks like it's simply an out of date comment, from what I can tell. Here's a [link to the PR](https://github.com/open-telemetry/opentelemetry-collector/pull/640) that included the change you've referenced. There was a bit of a [discussion](https://github.com/open-telemetry/opentelemetry-collector/pull/640#discussion_r413213888) on the value of anchoring the given regex, and it was agreed upon to not include a specific option for this. From this, I believe the anchor was removed intentionally, and the comment was just missed.

I've opened a PR to remove the incorrect comment.
